test/parallel_api/iterator/permutation_iterator.pass.cpp

```diff
@@ -35,6 +35,13 @@ main()
     const std::size_t perm_size_expected = kDefaultIndexStepOp.eval_items_count(countingItDistanceResult);
     auto permItBegin = dpl::make_permutation_iterator(countingItBegin, kDefaultIndexStepOp);
     auto permItEnd = permItBegin + perm_size_expected;
+
+    //transform_iterator is not trivially_copyable, as it defines a copy assignment operator which does not copy
```

Comment:
But transform_iterator has a default copy constructor. Is it enough to be a trivial copyable type.

(https://godbolt.org/z/KheKPYq37)

Reply:
I believe the situation we have is more like this:

https://godbolt.org/z/excn3oYdo

The presence of the eligible copy assignment operator which is non-trivial breaks `is_trivially_copyable`.

https://en.cppreference.com/w/cpp/language/classes#Trivially_copyable_class
https://en.cppreference.com/w/cpp/language/copy_assignment#Trivial_copy_assignment_operator

Reply:
Actually, I would merely remove 
`transform_iterator(const transform_iterator& __input) = default;`
and 
`transform_iterator& operator=(const transform_iterator& __input)`

It makes "transform_iterator" trivially copyable

Reply:
Discussed offline.  This creates a new requirement that lambdas are copy_assignable, which is not the case for lambdas with value captures (because those are represented as const member variables, and the copy assignment operator is then deleted).  
Without copy assignable functors, we then lose the ability to copy assign `transform_functor`. Which is a bigger problem than it not being "trivially copyable".  
Note that trivially copyable does not imply "copy assignable".

Reply:
In that case, as we discussed with Alexey, we should just define the trait sycl::is_device_copyable_v<T> to true for transform_iterator type.

Reply:
Yes, but we also discussed delaying that change to be addressed after the release, as it is not currently required to resolve the issues we are seeing.  I see that as an independent (although related) issue.  

If you insist on including this to approve this PR, then I will work on getting that in.  However, this is something we have not explored yet, and we need to merge today.